lib/auth/native/native.go

```diff
@@ -278,6 +284,19 @@ func (k *Keygen) GenerateUserCertWithoutValidation(c services.UserCertParams) ([
 		cert.Permissions.Extensions[teleport.CertExtensionAllowedResources] = c.AllowedResourceIDs
 	}
 
+	if c.SourceIP != "" {
+		if cert.CriticalOptions == nil {
+			cert.CriticalOptions = make(map[string]string)
+		}
+		//IPv4, all bits matter
+		ip := c.SourceIP + "/32"
+		if strings.Contains(c.SourceIP, ":") {
+			//IPv6
+			ip = c.SourceIP + "/64"
```

Comment:
IPv6 is 128 bits, not 64, right?

Reply:
of course it is, I don't know how I came up with 64